crates/project_panel/src/project_panel.rs

```diff
@@ -237,6 +238,12 @@ impl ProjectPanel {
                 project::Event::ActivateProjectPanel => {
                     cx.emit(PanelEvent::Activate);
                 }
+                project::Event::DiskBasedDiagnosticsFinished { .. }
+                | project::Event::DiagnosticsUpdated { .. } => {
+                    this.update_diagnostics(cx);
+                    this.update_visible_entries(None, cx);
+                    cx.notify();
+                }
```

Comment:
We could add a property in the settings to opt in or out of this feature and could not update the diagnostics in case someone does not want the feature.